code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DeploymentUnit.java

```diff
@@ -223,9 +226,34 @@ public void remove(String reason, String level) {
             instance.remove();
         }
 
-        // remove deployment unit object
         if (unit != null) {
-            context.objectProcessManager.scheduleStandardProcessAsync(StandardProcess.REMOVE, unit, null);
+            clenaupVolumes();
+            Map<String, Object> params = new HashMap<>();
+            params.put(ObjectMetaDataManager.REMOVED_FIELD, new Date());
+            params.put(ObjectMetaDataManager.REMOVE_TIME_FIELD, new Date());
```

Comment:
@ibuildthecloud I can't delete the row as it is being referenced by the volume object